src/pwrite.rs

```diff
@@ -39,29 +38,24 @@ where
     {
         self.pwrite_with(n, offset, Ctx::default())
     }
+
     /// Write `N` at offset `I` with context `Ctx`
     /// # Example
     /// ```
     /// use scroll::{Pwrite, Pread, LE};
     /// let mut bytes: [u8; 8] = [0, 0, 0, 0, 0, 0, 0, 0];
     /// bytes.pwrite_with::<u32>(0xbeefbeef, 0, LE).unwrap();
     /// assert_eq!(bytes.pread_with::<u32>(0, LE).unwrap(), 0xbeefbeef);
-    fn pwrite_with<N: TryIntoCtx<Ctx, <Self as Index<RangeFrom<usize>>>::Output, Error = E>>(
+    fn pwrite_with<N: TryIntoCtx<Ctx, Self::Buffer, Error = E>>(
         &mut self,
         n: N,
         offset: usize,
         ctx: Ctx,
-    ) -> result::Result<usize, E> {
-        let len = self.measure_with(&ctx);
-        if offset >= len {
-            return Err(error::Error::BadOffset(offset).into());
-        }
-        let dst = &mut self[offset..];
-        n.try_into_ctx(dst, ctx)
-    }
+    ) -> result::Result<usize, E>;
```

Comment:
so I don't like that this default impl is removed, but then I forgot there is a blanket impl of Pwrite below anyway, which is empty, because of this default impl; and this just gets moved there. The logic is a bit more separated, but it makes custom container types more flexible, which is quite cool.